dev-tools/reconfigurator-cli/src/lib.rs

```diff
@@ -2765,6 +2806,49 @@ fn cmd_save(
     )))
 }
 
+fn cmd_state_log(
+    sim: &mut ReconfiguratorSim,
+    args: StateLogArgs,
+) -> anyhow::Result<Option<String>> {
+    let StateLogArgs { from, limit, verbose } = args;
+
+    // Build rendering options.
+    let options = GraphRenderOptions::new(sim.current)
+        .with_verbose(verbose)
+        .with_limit(limit)
+        .with_from(from);
+
+    // Render the graph.
+    let output = sim.sim.render_graph(&options);
+
+    Ok(Some(output))
+}
+
+fn cmd_state_switch(
+    sim: &mut ReconfiguratorSim,
+    args: StateSwitchArgs,
+) -> anyhow::Result<Option<String>> {
+    // Try parsing as a full UUID first, then fall back to prefix matching.
+    let target_id = match args.state_id.parse::<ReconfiguratorSimUuid>() {
+        Ok(id) => id,
+        Err(_) => sim.sim.get_state_by_prefix(&args.state_id)?,
+    };
```

Comment:
Would it be worth making `args.state_id` a stronger type than string? I'm comparing it to something like `BlueprintIdOpt` and imagining

```rust
enum StateIdOpt {
    Id(ReconfiguratorSimUuid),
    IdPrefix(String),
}
```

but am not sure it's really worth it. I guess maybe it depends on how many other functions end up needing to repeat this "try to parse then treat as a prefix if we can't" logic.